src/appInsights/appInsightsWrapper.ts

```diff
@@ -0,0 +1,142 @@
+import { Context, HttpRequest } from '@azure/functions';
+import * as appInsights from 'applicationinsights';
+import { createAppInsightsLogger, consoleLogger } from './Logger';
+import { TelemetryClient } from 'applicationinsights';
+
+const telemetryClients: { [key: string]: TelemetryClient } = {};
+
+const isDisabled = process.env.ENVIRONMENT === undefined;
```

Comment:
Is `ENVIRONMENT` supposed to contain the AppInsight Key or just something like `ENVIRONMENT=true` or the deployed environment like `ENVIRONMENT=staging`  ?

Reply:
Currently the later. But is more or less a placeholder, till we have a better mechanism to enable/disable plugins. Currently the user has to do something like that:

```
if (isLocal) {
    middleware([headerAuthentication()], functionHandler, []);
else {
   middleware([AppInsightForHttpTrigger.setup, headerAuthentication()], functionHandler, [AppInsightForHttpTrigger.finalizeAppInsight]);
}
```

If we had a plugin-mechanism, we could do something like that:
(just a rough example)
```
if (!isLocal) {
    middleware.use(appinsightLogging())
}
```

Reply:
The setup function checks for the presence of the variable `ENVIRONMENT`. When this variable is not set local, the console logger is used. Correct me if I'm wrong but there is no additional logic for the local execution necessary ?
The used variable could be a bite more precise like `DEPLOYMENT_ENVIRONMENT` or something like that.

An additional configuration for the APP_INSIGHT Key is not necessary ?

Reply:
No no, either `APPINSIGHTS_INSTRUMENTATIONKEY` or `APPLICATIONINSIGHTS_CONNECTION_STRING` is necessary... You mean, we could use those two keys instead of process.env.ENVIRONMENT?

Reply:
Yes then the Environment-Variable `Environment` is not necessary, or ?

Reply:
Alright, I will change it 👍